### PR TITLE
GSoC 2018: Final Report - MagickPlot

### PR DESCRIPTION
**Pull request by arafatkatze**

### CODE REVIEW

**Review by prasunanand — APPROVED** [review]

File: source/_posts/2019-01-17-gsoc-2018-Ruby-Matplotlib.markdown [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+comments: true
+author: Arafat Dad Khan
+categories: [GSoC, GSoC 2018, Ruby, Plotting, Visualization, Matplotlib]
+---
+
+This is a wrap of my Gsoc project at Sciruby where I worked on builing Matplotlib for Ruby.
+
+At Sciruby, we came across the ambitious task of designing the next Matplotlib for Ruby and so we examined quite a few different plotting libraries and figured that if we could put together ideas from all of these libraries then we can make something really amazing.Primarily our main source of inspiration was Matplotlib. The Matplotlib Architecture is be broadly divided into three layers. The Backend, The Artist and the scripting layer.
+
+Matplotlib architecture is mostly written in Python with some of its backend (the renderer AGG and Figure canvas ) written in C++ ( The original AGG backend and helper scripts, which is quite tightly bound to python). But the recently the backends are also written in Python using the renderers which have Python APIs. The finer details of the Architecture can be found [here](https://www.aosabook.org/en/matplotlib.html).
```
Comment (v0dro) [inline source/_posts/2019-01-17-gsoc-2018-Ruby-Matplotlib.markdown]:
```suggestion
Matplotlib architecture is mostly written in Python with some of its backend (the renderer AGG and Figure canvas ) written in C++ ( The original AGG backend and helper scripts, which is quite tightly bound to python). But recently the backends are also written in Python using the renderers which have Python APIs. The finer details of the Architecture can be found [here](https://www.aosabook.org/en/matplotlib.html).
```

File: source/_posts/2019-01-17-gsoc-2018-Ruby-Matplotlib.markdown [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Future Work
+
+Our ultimate goal is to make this project similar to a matplotlib equivalent for Ruby with tons of really amazing customization features. I hope you find this interesting and useful.The library is currently being developed by the Sciruby community, feel free to try it out from [github](https://github.com/SciRuby/rubyplot/). Any suggestions and recommendations are welcome.
+
+## FOSS
+
+I have been an active contributor to a few open source projects and I have started a few
+nice ones of my own and I feel really glad to have been introduced to the open
+source community. I really appreciate the effort by Google Open Source Committee for conducting GSoC every year. It is the best platform for the aspiring programmers to improve their skill and give back to society by developing free and open source software.
```
Comment (v0dro) [inline source/_posts/2019-01-17-gsoc-2018-Ruby-Matplotlib.markdown]:
```suggestion
source community. I really appreciate the effort by Google Open Source Committee for conducting GSoC every year. It is the best platform for aspiring programmers to improve their skill and give back to society by developing free and open source software.
```

**Review by v0dro — CHANGES_REQUESTED** [review]

**Comment by prasunanand** [comment]
@Arafatk , Wrap it up. Make the changes.

**Comment by arafatkatze** [comment]
@prasunanand  Sorry for the delay.       
I hope this is okay.

**Pull request merged**